delegated-staking.md

```diff
@@ -0,0 +1,171 @@
+
+  ---
+
+### **Pending investigations**
+
+1. In the above-described logic, delegated funds count directly to the stakers *election weight* (proper term required). Staker's funds are required to be locked for a certain amout of epochs. This is diametral to the current delegation logic of allowing `delegated stakes` to be contributed and withdrawn at any time. 
+
+    For the contributing: Where is what kind of additional logic needed to somehow delay the attribution of the delegated funds to the staking power of a block producer, or, somehow otherwise take the necessity of a constant amount of funds to be locked for some epochs into account?
+
+    For the withdrawing: What checks and conditions could be applied to the `withdraw_delegated_stakes` function to prevent delegations to be withdrawn at an epoch they are supposed to be still locked, semantically as part of a locked block producer's stake?
+    
+2. Although `split_reward_to_delegators()` only distributes funds nominally instead of transfering actual value, it is to be verified whether the call to it has actually `value` assigned to it or not, therefore adding / removing the `payable` modifier might be necessary.
+  
+  
+### **Contextual constraints**
+ The above delegation logic assumes the following requirements to the `main staking logic` to be met: 
+ 
+ 1. On any adjustment of a staker's amount, who registered as a `delegatee`, `update_delegatees_stake()` is called to keep the delegation logic's book-keeping in sync with the amount of funds staked by stakers in the `main staking logic`. By checking if the Caller is a `delegatee`, all delegatee's calls to adjust their stake shall be forwarded *to the delegation logic* (as for example the direct withdrawal from the staker's total stake in the `main staking logic` would withdraw also the funds *delegated* to him). The delegation logic has means of adjusting the staked amount in the `main staking logic` in accordance to the changes in its book keeping, according stubs are included in the current code.
+ 3. It is assumed there is some function in the `main staking logic` that is called by the block producer every time he produces a block. (Current information: There is supposedly a function called `step()` to which this applies).
+ 4. It is assumed that rewards for blocks produced by stakers that registered as `delegatee`s remain in the contract of the `main staking logic`, from which the eligible parties can withdraw them utilizing the delegation logic's `withdraw_rewards()`. 
+ 5. Every time a block was produced, the `split_reward()` function is to be called (presumably in the function referenced in 3.) . This is safe, because `split_reward()` immediately returns, if the caller is not a registered `delegatee`. It is assumed, that in this call, the earned block reward is `Call.value`. If that is not the case, the delegation logic can easily be adjusted to reference this value from somewhere else which is accessible to the contract.
```

Comment:
At the end of an epoch - and with some (1 epoch?) delay - rewards are distributed. The StakingValidator interface allows a callback to be registered; this function will be called with information of rewards during the given epoch.